File: dev-docs/snippets/documents/commit-offline-create.ts
import type { JsBaoClient } from "js-bao-wss-client";

// Commit a locally-created (localOnly) document to the server.
export async function commitOfflineCreate(
  client: JsBaoClient,
  documentId: string,
) {
  // #region example
  const result = await client.documents.commitOfflineCreate(documentId, {
    onExists: "link",
  });
  // #endregion example
  return result; // { created, linked?, reason? }
}

File: dev-docs/snippets/documents/commit-offline-create.swift
import JsBaoClient

// Commit a locally-created (localOnly) document to the server. On Swift this is
// a top-level client method rather than a `documents.*` namespaced one.
func commitOfflineCreate(client: JsBaoClient, documentId: String) async throws {
  // #region example
  let result = try await client.commitOfflineCreate(
    documentId: documentId,
    onExists: "link"
  )
  // #endregion example
  _ = result // ["created": true] / ["linked": true] / ["reason": "exists"]
}
